### ISSUE TITLE
Impl Readable for Message

### ISSUE THREAD

**Issue by benthecarman** [p2pderivatives/rust-dlc#170]
`Readable` is implemented for `WireMessage` but not for `Message`, would be nice to have both options available

**Comment by Tibo-lg** [comment]
Can you explain a bit more what you need? I don't think `Readable` is implemented for `WireMessage` either. There is the `read_dlc_message` method in `message_handler` which then leverages the `Readable` implementation of the underlying messages but I think that's all?

**Comment by benthecarman** [comment]
Yeah, I'm trying to add my own transport mechanism (nostr) but having trouble because there's no easy way to read a Message. Like you said currently the only way is to have the MessageHandler read from the parsed TLV. Would be nice if I could just give it a byte buffer to read with Readable to parse the messages.

**Comment by Tibo-lg** [comment]
The `MessageHandler` implements `CustomMessageReader`, which takes a message type (`u16`) and then a buffer, and will return a `WireMessage`. So you could just read the first two bytes and then call this method to de-serialize. If you don't care about segmentation, you can either just not segment the messages when you write them, or implement your own `MessageHandler` that deals directly with `Message`. I guess it would be possible to implement `Readable` for `Message` using the type numbers defined in the specs but that removes quite some flexibility (e.g., using Nostr you probably can use different types).
